iib/workers/tasks/utils.py

```diff
@@ -175,28 +176,49 @@ def _requires_max_ocp_version(bundle):
     return False
 
 
-def serve_index_registry(db_path):
+def serve_registry(base_dir, from_index=None, fbc_dir=None, db_path=None):
```

Comment:
Since this has two code paths, I would recommend having two separate functions instead of clubbing them into one. That way it would also be easier to get rid of the old code once it becomes obsolete. 

You can probably make another helper for `opm serve/ opm registry serve` which will run the command based on a flag being passed.

Reply:
Also, if you do create the helper for that opm command, I would recommend putting it in a new module `opm_operations.py`. It might be a nice start to code cleaning. But I would leave this as optional

Reply:
Let me deeply thing about better solution. :thinking:

Reply:
so currently we don't have any use-case where fbc_dir parameter is used, is that for future cases? In that case should we add it here already?

Reply:
I agree that we are not using it now, we can remove `fbc_dir`

Reply:
@yashvardhannanavati Since I liked your ideas I did them... good luck with code-review :smile: